### PR TITLE
[#129915689]turn `_Line` utility class into more robust `_StringPipe` class

### PR DESCRIPTION
**Pull request by risicle**
Tangentially related to https://www.pivotaltracker.com/story/show/129915689

### CODE REVIEW

**Comment by TheDoubleK** [comment]
I'm not sure I see the benefit here?  Why might we want to `iter_csv` with something other than a csv writer? What does `_StringPipe` do for us that the previous code wasn't doing?

**Comment by risicle** [comment]
I was _going_ to use a `DictWriter` but it turned out I didn't in the end anyway, so much of that bit is unnecessary now.

`_StringPipe` (other than having a better name) doesn't flush its buffer every time it's written to, instead concatenating it to any existing input. `_Line` would work fine as long as your code was guaranteed to do `write` `read` `write` `read` `write` `read` `write` `read`... if you did two successive writes before a read you would end up with some of your stream missing.

**Comment by risicle** [comment]
Ok, updated to remove now-redundant flexibility-adding features. It's just the `_StringPipe` fix now.

File: tests/test_csv_generator.py [hunk 1]
```diff
@@ -5,11 +5,12 @@ class TestIterCsv():
     def test_it_creates_csv_lines_from_lit_of_rows(self):
```
Comment (TheDoubleK) [inline tests/test_csv_generator.py]: Just spotted typo in the test name `lit` -> `list`.  I reckon we can live with it for now if you can't face waiting for Travis again.

**Comment by TheDoubleK** [comment]
:+1:

File: tests/test_csv_generator.py (hunk 1 above)
Comment (risicle) [inline tests/test_csv_generator.py]: nah travis ain't too slow for utils

**Pull request merged**